documentation/docs/05-load.md

```diff
@@ -180,96 +227,79 @@ export type PageLoad = import('@sveltejs/kit').Load<{}>;
+export type LayoutServerLoad = import('@sveltejs/kit').ServerLoad<{}>;
 
 // @filename: index.js
 // ---cut---
 import { error } from '@sveltejs/kit';
 
-/** @type {import('./$types').PageLoad} */
-export async function load({ fetch }) {
-	const response = await fetch('https://secret.admin.stuff');
-	if (!response.ok) {
-		throw error(response.status, 'Something went wrong');
+/** @type {import('./$types').LayoutServerLoad} */
+export function load({ locals }) {
+	if (!locals.user) {
+		throw error(401, 'not logged in');
 	}
 
-	return {
-		article: await response.json()
-	};
+	if (!locals.user.isAdmin) {
+		throw error(403, 'not an admin');
+	}
 }
 ```
 
-#### redirect
+If an _unexpected_ error is thrown, SvelteKit will invoke [`handleError`](/docs/hooks#handleerror) and treat it as a 500 Internal Server Error.
 
-If the page should redirect (because the page is deprecated, or the user needs to be logged in, or whatever else), throw a `redirect` containing the location to which they should be redirected alongside a `3xx` status code.
+> In development, stack traces for unexpected errors are visible as `$page.error.stack`. In production, stack traces are hidden.
 
-```js
-/// file: src/routes/fast/+page.js
-// @filename: $types.d.ts
-export type PageLoad = import('@sveltejs/kit').Load<{}>;
+### Redirects
 
-// @filename: index.js
-// ---cut---
-import { redirect } from '@sveltejs/kit';
+Redirecting users works similarly:
 
-/** @type {import('./$types').PageLoad} */
-export async function load({ fetch }) {
-	throw redirect(301, '/faster');
+```diff
+/// file: src/routes/admin/+layout.server.js
+-import { error } from '@sveltejs/kit';
++import { error, redirect } from '@sveltejs/kit';
+
+/** @type {import('./$types').LayoutLoad} */
+export function load({ locals }) {
+	if (!locals.user) {
+-		throw error(401, 'not logged in');
++		throw redirect(307, '/login');
+	}
+
+	if (!locals.user.isAdmin) {
+		throw error(403, 'not an admin');
+	}
 }
 ```
 
-The `redirect` string should be a [properly encoded](https://developer.mozilla.org/en-US/docs/Glossary/percent-encoding) URI. Both absolute and relative URIs are acceptable. You can learn more about redirects and redirect codes [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Redirections).
+### Invalidation
+
+SvelteKit tracks the dependencies of each `load` function to avoid re-running it unnecessarily during navigation. For example, a `load` function in a root `+layout.js` doesn't need to re-run when you navigate from one page to another unless it references `url` or a member of `params` that changed since the last navigation.
+
+Using [`invalidate(url)`](/docs/modules#$app-navigation-invalidate), you can re-run any `load` functions that depend on the invalidated resource (either implicitly, via [`fetch`](#fetch)), or explicitly via [`depends`](#depends). You can also invalidate _all_ `load` functions by calling `invalidate()` without an argument.
```

Comment:
One reading this again: Do you mean to rerun the `load` functions even if they didn't fetch anything/didn't declare any `depends`?